ton_client/client/src/processing/process_message.rs

```diff
@@ -1,84 +1,98 @@
+                encode_params.processing_try_index = Some(try_index);
+                crate::abi::encode_message(context.clone(), encode_params)
+                    .await?
+                    .message
+            }
+        };
 
-// fn ensure_message(
-//     context: &Arc<ClientContext>,
-//     source: &MessageSource,
-//     retry_count: u32,
-//     callback: &Option<CallbackParams>,
-// ) -> ApiResult<(String, Option<u32>)> {
-//     Ok(match source {
-//         MessageSource::Message(boc) => (boc.clone(), None),
-//         MessageSource::AbiEncoding(encode_params) => {
-//             emit_event(context, callback, || MessageProcessingEvent::EncodeMessage);
-//             let encoded = crate::abi::encode_message(context.clone(), encode_params.clone())?;
-//             (encoded.message, None)
-//         }
-//     })
-// }
-//
-// pub async fn process_message(
-//     context: Arc<ClientContext>,
-//     params: ParamsOfProcessMessage,
-// ) -> ApiResult<ResultOfProcessMessage> {
-//     let mut retry_count = 0;
-//     loop {
-//         let (message, expiration_time) =
-//             ensure_message(&context, &params.message, retry_count, &params.callback)?;
-//         let transaction_waiting_state = send_message(
-//             context.clone(),
-//             ParamsOfSendMessage {
-//                 message: message.clone(),
-//                 message_expiration_time: None,
-//                 callback: params.callback.clone(),
-//             },
-//         )
-//         .await?
-//         .transaction_waiting_state;
-//
-//         let result = wait_for_transaction(
-//             context.clone(),
-//             ParamsOfWaitForTransaction {
-//                 message: message.clone(),
-//                 message_expiration_time: expiration_time,
-//                 callback: params.callback.clone(),
-//                 waiting_state: transaction_waiting_state,
-//             },
-//         )
-//         .await?;
-//         match result {
-//             ResultOfWaitForTransaction::Complete(transaction) => {
-//                 emit_event(&context, &params.callback, || {
-//                     MessageProcessingEvent::TransactionReceived {
-//                         transaction: transaction.clone(),
-//                     }
-//                 });
-//                 return Ok(ResultOfProcessMessage { transaction });
-//             }
-//
-//             ResultOfWaitForTransaction::Incomplete(waiting_state) => {}
-//         }
-//         retry_count += 1;
-//     }
-// }
+        // Send
+        let mut processing_state = send_message(
+            context.clone(),
+            ParamsOfSendMessage {
+                message: message.clone(),
+                abi: abi.clone(),
+                callback: params.callback.clone(),
+            },
+        )
+        .await?
+        .processing_state;
+
+        // Monitor network
+        loop {
+            match wait_for_transaction(
```

Comment:
Make variable with result and use it in match. It's hard to understand this construction

Reply:
done